### ISSUE TITLE
Namespace table struct and methods

### ISSUE THREAD

**Issue by philippecamacho** [EspressoSystems/espresso-sequencer#1056]
[Currently](https://github.com/EspressoSystems/espresso-sequencer/blob/dc150012ad9bc55a63f4b97138155e545b75397e/sequencer/src/block2.rs#L442) the logic of the namespace table is spread across several functions which makes it hard to follow / improve.
Create a new struct and related methods that capture the behaviour of a namespace table.

**Comment by ggutoski** [comment]
subsumed by #1076
